src/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -63,7 +63,8 @@ public static ConsoleColor BackgroundColor
 
         public static void ResetColor()
         {
-            if (!ConsoleOutIsTerminal)
+            // We only want to reset colors if we're targeting a TTY device
+            if (Console.IsOutputRedirected)
```

Comment:
You are right. I was still in the windows world where Console.SetOut() does not affect the IsXXXRedirected. I will change this to IsOutputRedirectedCore().